com.microsoft.mrtk.accessibility/Subsystems/MRTK/MRTKAccessibilitySubsystem.cs

```diff
@@ -44,6 +45,142 @@ public AccessibilityProvider() : base()
+
+            private readonly Dictionary<DescribableObjectClassification, List<GameObject>> describableObjects =
+                new Dictionary<DescribableObjectClassification, List<GameObject>>();
+
+            /// <inheritdoc/>
+            internal override bool TryGetDescribableObjects(List<DescribableObjectClassification> classifications, DescribableObjectVisibility visibility, float maxDistance, List<GameObject> describableObjectsList)
+            {
+                if (maxDistance <= 0)
+                {
+                    Debug.LogError("The distance from the camera to the objects cannot be less than or equal to zero.");
+                    return false;
+                }
+
+                describableObjectsList.Clear();
+
+                AssembleDescribableObjects(classifications, describableObjectsList);
+                FilterDescribableObjects(visibility, maxDistance, describableObjectsList);
+
+                return true;
+            }
+
+            /// <inheritdoc/>
+            public override bool TryGetDescribableObjectClassifications(List<DescribableObjectClassification> classifications)
+            {
+                classifications.Clear();
+                classifications.AddRange(describableObjects.Keys);
+                return true;
+            }
+
+            /// <inheritdoc/>
+            public override bool TryRegisterDescribableObject(GameObject describableObject, DescribableObjectClassification classification)
+            {
+                // Make sure the specified game object has not been previously registered in any classification.
+                foreach (IList<GameObject> list in describableObjects.Values)
+                {
+                    if (list.Contains(describableObject))
+                    {
+                        Debug.LogError($"{describableObject.name} has already been registered as a describable object");
+                        return false;
+                    }
+                }
+
+                if (!describableObjects.ContainsKey(classification))
+                {
+                    describableObjects.Add(classification, new List<GameObject>());
+                }
+                describableObjects[classification].Add(describableObject);
+
+                return true;
+            }
+
+            /// <inheritdoc/>
+            public override bool TryUnregisterDescribableObject(GameObject describableObject, DescribableObjectClassification classification)
+            {
+                List<GameObject> objCollection = describableObjects[classification];
+                if (!objCollection.Contains(describableObject))
+                {
+                    Debug.LogError($"{describableObject.name} has not been registered as a describable object of classification {classification}");
+                    return false;
+                }
+
+                if (!describableObjects[classification].Remove(describableObject))
```

Comment:
Can this reference the previously-cached `objCollection` so we don't query a second time for the same key?
```suggestion
                if (!objCollection.Remove(describableObject))
```